park-dashboard/src/Components/AirQuality/AirQuality.jsx

```diff
@@ -1 +1,125 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import "./AirQuality.css";
+import ReactSpeedometer from "react-d3-speedometer";
+
+function AirQuality(props) {
+  const [aqi, setAqi] = useState(0);
+  const [condition, setCondition] = useState();
+  const [description, setDescription] = useState();
+  const [color, setColor] = useState();
+
+  const parkCode = props.parkCode;
+
+  const getAQIConditionDescriptionAndColor = (aqi) => {
+    if (aqi >= 0 && aqi <= 50) {
+      return {
+        condition: "Good (0-50)",
+        description:
+          "Air quality is satisfactory, and there is little to no health risk.",
+        color: "#39b54930",
+      };
+    } else if (aqi > 50 && aqi <= 100) {
+      return {
+        condition: "Moderate (51-100)",
+        description:
+          "Air quality is acceptable, but certain pollutants may moderately affect sensitive individuals.",
+        color: "#ffff0050",
+      };
+    } else if (aqi > 100 && aqi <= 150) {
+      return {
+        condition: "Unhealthy for Sensitive Groups (101-150)",
+        description:
+          "Some individuals, particularly those with respiratory or heart conditions, may experience health issues due to the elevated pollution levels.",
+        color: "#f05a2250",
+      };
+    } else if (aqi > 150 && aqi <= 200) {
+      return {
+        condition: "Unhealthy (151-200)",
+        description:
+          "The general population may experience health effects, and sensitive individuals' conditions may worsen.",
+        color: "#ff000050",
+      };
+    } else if (aqi > 200 && aqi <= 300) {
+      return {
+        condition: "Very Unhealthy (201-300)",
+        description:
+          "The air quality is significantly compromised, and there is a higher risk of respiratory problems and other health issues for the general population.",
+        color: "#64017950",
+      };
+    } else {
+      return {
+        condition: "Hazardous (>300)",
+        description:
+          "Air quality is severely polluted, posing a significant risk to everyone's health, not just those with pre-existing conditions.",
+        color: "#9e010350",
+      };
+    }
+  };
+
+  const fetchAirQuality = (latitude, longitude) => {
```

Comment:
If for some reason API call fails or we are not getting the Air quality data, do you think some error message can be added? Maybe something like "Sorry, the air quality couldn't be fetched currently"

Reply:
I added a default message for the air quality component.